src/database/removeRemote.ts

```diff
@@ -46,109 +31,99 @@ export class RTDBRemoveRemote implements RemoveRemote {
-          }
-          let data = {};
-          try {
-            data = JSON.parse(body);
-          } catch (e) {
-            return reject(
-              new FirebaseError("Malformed JSON response in shallow get ", {
-                exit: 2,
-                original: e,
-              })
-            );
-          }
-          if (data) {
-            const keyList = Object.keys(data);
-            return resolve(keyList);
-          }
-          resolve([]);
+      `.json?shallow=true&limitToFirst=${numChildren}`;
+    const t0 = Date.now();
+    return api
+      .addRequestHeaders({
+        url,
+      })
+      .then((reqOptionsWithToken) => {
+        return new Promise<string[]>((resolve, reject) => {
+          request.get(reqOptionsWithToken, (err: Error, res: Response, body: any) => {
+            if (err) {
+              return reject(
+                new FirebaseError("Unexpected error while listing subtrees", {
+                  exit: 2,
+                  original: err,
+                })
+              );
+            } else if (res.statusCode >= 400) {
+              return reject(responseToError(res, body));
+            }
+            let data = {};
+            try {
+              data = JSON.parse(body);
+            } catch (e) {
+              return reject(
+                new FirebaseError("Malformed JSON response in shallow get ", {
+                  exit: 2,
+                  original: e,
+                })
+              );
+            }
+            if (data) {
+              const keyList = Object.keys(data);
+              return resolve(keyList);
+            }
+            resolve([]);
+          });
         });
+      })
+      .then((paths: string[]) => {
+        const dt = Date.now() - t0;
+        logger.debug(`[database] Sucessfully fetched ${paths.length} path at ${path} ${dt}`);
+        return paths;
       });
+  }
+
+  private patch(path: string, body: any, note: string): Promise<boolean> {
+    const t0 = Date.now();
+    return new Promise((resolve, reject) => {
+      const url =
+        utils.addSubdomain(api.realtimeOrigin, this.instance) +
+        path +
+        ".json?print=silent&writeSizeLimit=tiny";
+      return api
+        .addRequestHeaders({
+          url,
+          body,
+          json: true,
+        })
+        .then((reqOptionsWithToken) => {
+          request.patch(reqOptionsWithToken, (err: Error, res: Response, resBody: any) => {
+            if (err) {
+              return reject(
+                new FirebaseError(`Unexpected error while removing data at ${path}`, {
+                  exit: 2,
+                  original: err,
+                })
+              );
+            }
+            const dt = Date.now() - t0;
+            if (res.statusCode >= 400) {
+              logger.debug(`[database] Failed to remove data at ${path} (${note}) time: ${dt}`);
+              return resolve(false);
+            } else {
```

Comment:
`else` is redundant since the if block above returns. (remove the `else` block and just do the two things inside eit)

Reply:
Done